reconcile/utils/oc.py

```diff
@@ -439,6 +440,26 @@ def get_service_account_username(user):
         name = user.split('/')[1]
         return "system:serviceaccount:{}:{}".format(namespace, name)
 
+    def recycle_orphan_pods(self, dry_run, namespace, resource):
+        pods = self.get(namespace, 'Pod')['items']
+        for p in pods:
+            owner = self.get_obj_root_owner(namespace, p)
+            if resource.kind != owner['kind']:
+                continue
+            if resource.name != owner['metadata']['name']:
+                continue
+            # we found a pod we want to recycle
+            name = p['metadata']['name']
+            self.delete(namespace, 'Pod', name)
+            self.validate_pod_ready(namespace, name)
+
+    @retry(max_attempts=20)
+    def validate_pod_ready(self, namespace, name):
+        pod = self.get(namespace, 'Pod', name)
+        for status in pod['status']['containerStatuses']:
+            if not status['ready']:
+                raise PodNotReadyError(name)
```

Comment:
If we arrive here, it means that we're bound to manual intervention. I would write a nice and long message describing the situation for the poor soul that has to fix this :)

Reply:
this is raised in the context of a deployment. so before this happens, we will see a log that a StatefulSet is being applied. i've added more logging to indicate that we are deleting and applying, and more logging to indicate that we are recycling pods and validating them. so the exception alone will not say much, but it will within the context.

and just a nit: we will reach here multiple times but will only raise it after the retry max attempts.

Reply:
As long as we know that we're recycling statefulsets pods in the context of an immutable change and which pod we're recycling I'm fine.

Reply:
:kissing: